src/utils/bouquetGroups.ts

```diff
@@ -0,0 +1,143 @@
+import { type ComputedRef, type Ref } from 'vue'
+
+import type { DatasetProperties, DatasetsGroups } from '@/model/topic'
+
+export const noGroup = 'Sans regroupement'
+
+export function useGroups(datasetsProperties: Ref<DatasetProperties[]>): {
+  groupedDatasets: ComputedRef
+  getDatasetIndex: (group: string, indexInGroup: number) => number
+  removeDatasetFromGroup: (group: string, index: number) => DatasetProperties[]
+  groupAlreadyExists: (groupName: string) => boolean
+  renameGroup: (oldGroupName: string, newGroupName: string) => void
+  deleteGroup: (groupName: string) => DatasetProperties[]
+} {
+  const groupedDatasets = computed(() => {
+    const datasetsGroups: Ref<DatasetsGroups> = ref(new Map())
+    // create the key for empty group
+
+    datasetsGroups.value.set(noGroup, [])
+
+    // Loop through the datasets and group them by the 'group' property
+    datasetsProperties.value.forEach((dataset) => {
+      if (dataset.group) {
+        if (!datasetsGroups.value.has(dataset.group)) {
+          // create the key for new a group with an empty array
+          datasetsGroups.value.set(dataset.group, [])
+        }
+        // push the dataset to the array of its group
+        datasetsGroups.value.get(dataset.group)?.push(dataset)
+      } else {
+        // push datasets without group to the empty key
+        datasetsGroups.value.get(noGroup)?.push(dataset)
+      }
+    })
+
+    // get all entries within nogroup if any exists
+    const noGroupEntries: DatasetProperties[] | undefined =
+      datasetsGroups.value.has(noGroup)
+        ? datasetsGroups.value.get(noGroup)
+        : undefined
+
+    // create a new Map without nogroup
+    const filteredGroups: DatasetsGroups = new Map(
+      [...datasetsGroups.value].filter(([key]) => key !== noGroup)
+    )
+
+    // Alphabetically sort the keys with nogroup at the end
+    const sortedEntries = Array.from(filteredGroups.entries()).sort(
+      ([keyA], [keyB]) => {
+        return keyA.toString().localeCompare(keyB.toString())
+      }
+    )
+
+    // Add the nogroup to the end of the sorted array
+    if (noGroupEntries) {
+      sortedEntries.push([noGroup, noGroupEntries])
+    }
+
+    return new Map(sortedEntries)
```

Comment:
```suggestion
    const sortedGroups = new Map(
      [...datasetsGroups.value.entries()]
        .sort(([keyA], [keyB]) => {
          if (keyA === noGroup) return 1
          if (keyB === noGroup) return -1
          return keyA.localeCompare(keyB)
        })
    )

    return sortedGroups
```

J'avoue, l'IA m'a un peu aidé sur ce coup là 😅 Je n'arrivais pas à le faire en aussi peu de ligne. Testé rapidement, m'a l'air pas mal.

Reply:
👀 

```js
  const groupedDatasets = computed(() => {
    // Group datasets by their group property
    const groupedMap = datasetsProperties.value.reduce((acc, dataset) => {
      const groupKey = dataset.group || NO_GROUP
      if (!acc.has(groupKey)) {
        acc.set(groupKey, [])
      }
      acc.get(groupKey)?.push(dataset)
      return acc
    }, new Map<string, DatasetProperties[]>())

    // Extract and remove NO_GROUP entries
    const noGroupEntries = groupedMap.get(NO_GROUP)
    groupedMap.delete(NO_GROUP)

    // Sort groups alphabetically
    const sortedEntries = Array.from(groupedMap.entries()).sort(
      ([keyA], [keyB]) => keyA.localeCompare(keyB)
    )

    // Add noGroup entries at the end if they exist
    if (noGroupEntries?.length) {
      sortedEntries.push([NO_GROUP, noGroupEntries])
    }

    return new Map(sortedEntries)
  })
```